src/kademlia/dht_tracker.cpp

```diff
@@ -644,6 +648,17 @@ namespace libtorrent { namespace dht {
 
 	bool dht_tracker::send_packet(aux::session_listen_socket* s, entry& e, udp::endpoint const& addr)
 	{
+		// this check is necessary because this function is triggered by the m_refresh_timer
+		// via dht.tick, if the pair (socket, node) is removed from the map while this operation
+		// is right to go executed, this address becomes invalid
+		if (m_nodes.find(s) == m_nodes.end()) // socket already removed
+		{
+#ifndef TORRENT_DISABLE_LOGGING
+			m_log->log(dht_logger::tracker, "ignoring send_packet due to removed listen socket");
+#endif
+			return false;
+		}
+
```

Comment:
This check is unnecessary because `dht_tracker::refresh_timeout` calls tick by iterating over `m_nodes` so the socket is guaranteed to be in `m_nodes`. It might be good to keep this as an assert though.